Source/WTF/wtf/PlatformHave.h

```diff
@@ -971,6 +971,7 @@
 #define HAVE_CFNETWORK_NSURLSESSION_HSTS_WITH_UNTRUSTED_ROOT 1
 #define HAVE_NSURLSESSION_TASK_DELEGATE 1
 #define HAVE_AUDIO_OBJECT_PROPERTY_ELEMENT_MAIN 1
+#define HAVE_MAIN_PORT 1
```

Comment:
Based on the API declaration for `IOMainPort`, I think you should do something like this:

```
#if (PLATFORM(MAC) && __MAC_OS_X_VERSION_MIN_REQUIRED >= 120000) \
    || PLATFORM(IOS_FAMILY)
#define HAVE_IOKIT_MAIN_PORT 1
#endif
```

Reply:
This matches exactly the ifdef above it though

Reply:
We build WebKit on the current OS (Ventura, 130000) and two down-level OSes (Monterey, 120000 and Big Sur, 110000). Big Sur does not have IOKitMain, and so your change will cause a build failure on that platform. Please use the definition I suggested to you.